tests/components/onewire/test_entity_owserver.py

```diff
@@ -0,0 +1,204 @@
+                "injected_value": b"    25.123",
+                "result": "25.1",
+                "unit": TEMP_CELSIUS,
+            },
+        ],
+    },
+    "EF.111111111112": {
+        "inject_reads": [
+            b"HB_MOISTURE_METER",  # read type
+            b"         1",  # read is_leaf_0
+            b"         1",  # read is_leaf_1
+            b"         0",  # read is_leaf_2
+            b"         0",  # read is_leaf_3
+        ],
+        "sensors": [
+            {
+                "entity_id": "sensor.ef_111111111112_wetness_0",
+                "unique_id": "/EF.111111111112/moisture/sensor.0",
+                "injected_value": b"    41.745",
+                "result": "41.7",
+                "unit": PERCENTAGE,
+            },
+            {
+                "entity_id": "sensor.ef_111111111112_wetness_1",
+                "unique_id": "/EF.111111111112/moisture/sensor.1",
+                "injected_value": b"    42.541",
+                "result": "42.5",
+                "unit": PERCENTAGE,
+            },
+            {
+                "entity_id": "sensor.ef_111111111112_moisture_2",
+                "unique_id": "/EF.111111111112/moisture/sensor.2",
+                "injected_value": b"    43.123",
+                "result": "43.1",
+                "unit": PRESSURE_CBAR,
+            },
+            {
+                "entity_id": "sensor.ef_111111111112_moisture_3",
+                "unique_id": "/EF.111111111112/moisture/sensor.3",
+                "injected_value": b"    44.123",
+                "result": "44.1",
+                "unit": PRESSURE_CBAR,
+            },
+        ],
+    },
+}
+
+
+@pytest.mark.parametrize("device_id", MOCK_DEVICE_SENSORS.keys())
+async def test_owserver_setup_valid_device(hass, device_id):
+    """Test for 1-Wire device."""
+    entity_registry = mock_registry(hass)
+
+    dir_return_value = [f"/{device_id}/"]
+    read_side_effect = [device_id[0:2].encode()]
+    if "inject_reads" in MOCK_DEVICE_SENSORS[device_id]:
+        read_side_effect += MOCK_DEVICE_SENSORS[device_id]["inject_reads"]
+
+    expected_sensors = MOCK_DEVICE_SENSORS[device_id]["sensors"]
+    for expected_sensor in expected_sensors:
+        read_side_effect.append(expected_sensor["injected_value"])
+
+    with patch("homeassistant.components.onewire.sensor.protocol.proxy") as owproxy:
+        owproxy.return_value.dir.return_value = dir_return_value
+        owproxy.return_value.read.side_effect = read_side_effect
+
+        assert await async_setup_component(hass, SENSOR_DOMAIN, MOCK_CONFIG)
+        await hass.async_block_till_done()
+
+    assert len(entity_registry.entities) == len(expected_sensors)
+
+    for expected_sensor in expected_sensors:
+        entity_id = expected_sensor["entity_id"]
+        registry_entry = entity_registry.entities.get(entity_id)
+        assert registry_entry is not None
+        assert registry_entry.unique_id == expected_sensor["unique_id"]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)